### PR TITLE
Portfolio: details popup window

### PR DESCRIPTION
**Pull request by Mugiira-Kim**
**In this project we were able to complete the following requirements.**

When the user clicks (or taps) the button to check project details, the popup with details about the project appears.
When the user clicks (or taps) the close (X) button, the popup disappears.
Stuck to the Figma design as much as possible.
Used JavaScript array to store information for all projects.
Updated the main page so that the projects section is created dynamically using the information stored in that JavaScript object.

All the learning objectives were met as directed.

### CODE REVIEW

File: windowpopup.js [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+
+const mobileDummyText = 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.';
```
Comment (OLIPLICHE) [inline windowpopup.js:61]: - [ ] Please, make sure that when we click on "See project" bottom, the popup with details about the project must be the same on the mobile and desktop versions.

File: windowpopup.js [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+    description: 'A daily selection of privately personalized reads; no accounts or sign-ups required.',
+    skills: ['html', 'Ruby on rails', 'css', 'javascript'],
+    skillsModal: ['html', 'css', 'javascript', 'github', 'ruby', 'Bootstraps'],
+    livelink: { link: 'felix45.github.io/portfolio', text: 'See live' },
+    seesource: { link: 'github.com/felix45/portfolio', text: 'See Source' },
+    seeproject: 'See Project',
+    htmlClass: { article: ['card', 'article-card', 'mb-3', 'm-sm-auto', 'p-sm-2', 'hide-ruby'], cardImage: ['col-md-6', 'images'], cardDescription: ['col-md-6'] },
+    rowClass: { row: ['row', 'g-0', 'p-2'] },
+  },
+
```
Comment (OLIPLICHE) [inline windowpopup.js:15]: - [ ] Please, the popup must be responsive, it fits mobile and desktop sizes properly.

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Linter is green (the last commit have a green check mark (✅)) 👏
- README file is descriptive 👏
- Commit messages are meaningful 👏 👏
- Github flow is correctly used (Pull Request is made from the feature branch into the main branch) 👏

## Required Changes ♻️

- [ ] Please change your ReadMe template. Kindly use this [README template](https://github.com/microverseinc/readme-template/blob/master/README.md) from Microverse

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: windowpopup.js (hunk 1 above)
Comment (Mugiira-Kim) [reply windowpopup.js:61]: @OLIPLICHE  The desktop version on Figma seems to have more text than the mobile version.

Comment (sja-thedude) [reply windowpopup.js:61]: Good job in implementing the changes 👍

File: windowpopup.js (hunk 2 above)
Comment (sja-thedude) [reply windowpopup.js:15]: As suggested by the previous reviewer , kindly make sure that the popup is responsive and fits properly to mobile and desktop sizes. For this you have to do the following:-

Mobile Popup:

- [ ] Kindly make sure that the popup is of the same height and width as per the Figma design. As you can see that your popup heading is not visible. Do fix it.
- [ ] Optional: Kindly make sure that the background of the popup is blurred as per the Figma design.
- [ ] Optional: Kindly make sure to follow the font size for the paragraph as per the given design.
- [ ] Kindly follow the colors, spacing, and font for the entire popup and replicate as much as possible.

Figma design:



Your design:



Desktop Popup:

- [ ] Kindly make sure that the popup is of the same height and width as per the Figma design.
- [ ] Optional: Kindly make sure that the background of the popup is blurred as per the Figma design.
- [ ] Optional: Kindly make sure to follow the font size for the paragraph as per the given design.
- [ ] Optional: Kindly make sure to add bullets to the list above image as per the Figma design.
- [ ] Kindly follow the colors, spacing, and font for the entire popup and replicate as much as possible.

Your design:



Figma design:

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @Mugiira-Kim, This is @sja-thedude your code reviewer.

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

200w

 ## Highlights
✔️ Good job so far.

 ## Required Changes ♻️
- [ ] _Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: windowpopup.js (hunk 2 above)
Comment (rotimiazeez) [reply windowpopup.js:15]: > As suggested by the previous reviewer , kindly make sure that the popup is responsive and fits properly to mobile and desktop sizes. For this you have to do the following:-
> 
> Mobile Popup:
> 
> * [ ]  Kindly make sure that the popup is of the same height and width as per the Figma design. As you can see that your popup heading is not visible. Do fix it.
> * [ ]  Optional: Kindly make sure that the background of the popup is blurred as per the Figma design.
> * [ ]  Optional: Kindly make sure to follow the font size for the paragraph as per the given design.
> * [ ]  Kindly follow the colors, spacing, and font for the entire popup and replicate as much as possible.

Please note that you still need to work on this as suggested by the previous reviewer, to make your page UI (user interface) better and to match the original Figma design💪🏽👍

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
# Hi Team 💪🏽, 

### Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: windowpopup.js (hunk 2 above)
Comment (Mugiira-Kim) [reply windowpopup.js:15]: @rotimiazeez Kindly help me understand what you mean by doesn't fit screen. It is displaying perfectly on my side.The popup on your portfolio too does not fit on my screen, I don't get why you're all flagging a 'mistake' yourselves are unable to implement.

Comment (rotimiazeez) [reply windowpopup.js:15]: @Mugiira-Kim Kindly check the requirement and original design to make sure your design is a match.😉
Your design | Original design
screenshot (46) | screenshot (48).

Note that you can click on each element to see the style properties e.g font-family, font-size, color, width, height ... on the right side of your screen as seen in the screenshot.👍

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**